toolchain/check/deduce.cpp

```diff
@@ -119,31 +119,16 @@ class DeductionWorklist {
            context_->type_blocks().Get(args), needs_substitution);
   }
 
-  auto AddAll(SemIR::FacetTypeId params, SemIR::FacetTypeId args,
-              bool needs_substitution) -> void {
-    const auto& param_impls =
-        context_->facet_types().Get(params).impls_constraints;
-    const auto& arg_impls = context_->facet_types().Get(args).impls_constraints;
-    // TODO: Decide whether to error on these or just treat the parameter list
-    // as non-deduced. For now we treat it as non-deduced.
-    if (param_impls.size() != 1 || arg_impls.size() != 1) {
-      return;
-    }
-    auto param = param_impls.front();
-    auto arg = arg_impls.front();
-    if (param.interface_id != arg.interface_id) {
-      return;
-    }
-    Add(param.specific_id, arg.specific_id, needs_substitution);
-  }
-
   // Adds a (param, arg) pair for an instruction argument, given its kind.
   auto AddInstArg(SemIR::Inst::ArgAndKind param, int32_t arg,
                   bool needs_substitution) -> void {
     CARBON_KIND_SWITCH(param) {
       case SemIR::IdKind::None:
       case SemIR::IdKind::For<SemIR::ClassId>:
       case SemIR::IdKind::For<SemIR::IntKind>:
+      // Decided on 2025-04-02 not to do facet type deduction.
```

Comment:
Can the rationale be written down somewhere rather than just a decision?

Reply:
Expanded comment and updated doc.